dlx2.py

```diff
@@ -0,0 +1,178 @@
+class Node:
```

Comment:
I didn't know what DLX stood for until I saw [this article](https://gieseanw.wordpress.com/2011/06/16/solving-sudoku-revisited/) which explains it means "Dancing Links". Might be good to name the file with the whole word `dancing_links2.py`. You'll thank yourself someday when you look back at your code.